testing/build.gradle

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+ext {
+    noPublish = true
+}
+dependencies {
+    implementation project(':processor')
+    implementation project(':protobuf')
+
+    implementation "org.apache.kafka:kafka-clients:$kafkaVersion"
+    implementation "org.apache.kafka:kafka_2.12:$kafkaVersion"
+    implementation "org.apache.kafka:kafka-clients:$kafkaVersion:test"
+    implementation "org.apache.kafka:kafka_2.12:$kafkaVersion:test"
+
+    // zookeeper also brings slf4j-log4j12 and log4j dependencies
```

Comment:
if we expect it to be there (I think so as I see log4j.properties) at running test, let's not rely on transitive dependencies.

exclude them from this dep and add direct dep with controlled version is preferred.

Reply:
I included log4j to make debug easier when integration test fails.

Agree, I'll add dependency explicitly.